components/subcomponents/movieNight/upcomingMovie.py

```diff
@@ -0,0 +1,211 @@
+    if not upcoming_host_name:
+        embed.title = "You are not the upcoming movie night host!"
+        embed.description = "Upcoming movie night host has not been selected yet."
+        embed.description += "\nPlease contact a dictator so that they can select a host."
+    elif member_name != upcoming_host_name:
+        embed.title = "You are not the upcoming movie night host!"
+        embed.description = "{upcoming_host_name} is the upcoming movie night host."
+        embed.description += "\nPlease contact a dictator if you think there has been a mixup."
+    elif not (movie := suggestion_database.get_movie(member_name, movie_name)):
+        embed.title = f"{movie_name} does not exist in your suggestion list!"
+        embed.description = "Please add the movie to your suggestion list and then try again."
+    else:
+        db["upcoming_movie"] = movie
+        embed.colour = ut.embed_colour["MOVIE_NIGHT"]
+        embed.title = f"{member_name} finally picked a movie!"
+        embed.description = f"Next movie set as {movie.name}"
+        embed.description += f"\nThe movie will be watched on {ut.convert_to_ottawa_time(event.start_time)}."
+
+    db.close()
+    update_event_description(True)
+    return embed
+
+
+# Gets more details on the upcoming movie night
+# Returns str to be sent as message if successful
+# Otherwise, returns an embed that can be sent with a message
+def get_upcoming() -> discord.Embed | str:
+    # Should only be performed if ScheduledEvent MOVIE_EVENT_NAME exists
+    if failed_embed := ut.movie_event_not_present(True):
+        return failed_embed
+
+    return f"[Click here for more details.]({ut.get_movie_event_link()})"
+
+
+# Sends reminder to upcoming_host every noon until a movie from suggestion-list is picked
+def start_pick_reminder():
+    if __should_send_reminder():
+        pickReminderTask.start()
+
+
+# Updates the description of ScheduledEvent
+def update_event_description(is_command):
+    updateEventDescriptionTask.start(is_command)
+
+
+# Check to see if host should be reminded to pick a movie for movie night
+# Only set up reminders if:
+#   1) host has been picked, but not the movie
+#   2) ScheduledEvent MOVIE_EVENT_NAME exists
+#   3) ScheduledEvent takes place in the future
+def __should_send_reminder():
+    event: discord.ScheduledEvent | None = ut.get_movie_event()
+    db = shelve.open(UPCOMING_MOVIE_NIGHT_DB_PATH)
+    upcoming_host_name: str | None = db.get("upcoming_host_name")
+    upcoming_movie: Movie | None = db.get("upcoming_movie")
+    db.close()
+
+    return upcoming_host_name and not upcoming_movie and event and datetime.now(timezone.utc) <= event.start_time
+
+
+# Coroutine to remind the host to pick the movie
+# Runs in the background as a task in the following situations:
+#   1) every time an upcoming_host has been set
+#   2) every time the application restarts
+async def __remind_host_coroutine():
+    try:
+        while True:
```

Comment:
question: Why is there a while True here? Is this needed?

Reply:
This is an event reminder that sends a message every day until someone picks a movie. As such, the while loop is only exited if the host picks a movie (among other conditions).

The task runs in a separate thread that sleeps so it should not introduce any bugs related to hogging performance.